craft_parts/utils/os_utils.py

```diff
@@ -182,6 +184,39 @@ def is_snap(*, application_name: Optional[str] = None) -> bool:
     return res
 
 
+def get_system_info() -> str:
+    """Obtain running system information."""
+    # Use subprocess directly here. common.run_output will use binaries out
+    # of the snap, and we want to use the one on the host.
+    try:
+        output = subprocess.check_output(
+            [
+                "uname",
+                "--kernel-name",
+                "--kernel-release",
+                "--kernel-version",
+                "--machine",
+                "--processor",
+                "--hardware-platform",
+                "--operating-system",
+            ]
+        )
+    except subprocess.CalledProcessError as err:
+        logger.warning(
+            "'uname' exited with code %d: unable to record machine manifest",
+            err.returncode,
+        )
+        return ""
+
+    try:
+        uname = output.decode(sys.getfilesystemencoding()).strip()
+    except UnicodeEncodeError:
+        logger.warning("Could not decode output for 'uname' correctly")
+        uname = output.decode("latin-1", "surrogateescape").strip()
```

Comment:
I don't believe we need this anymore since 16.04 (14.04 being where this was problematic), nothing wrong with leaving it here though